odoo/addons/base/models/ir_attachment.py

```diff
@@ -483,23 +483,23 @@ def _check_access(self, operation):
         model_ids = defaultdict(set)            # {model_name: set(ids)}
         att_model_ids = []                      # [(att_id, (res_model, res_id))]
         # DLE P173: `test_01_portal_attachment`
-        remaining.flush_recordset(['res_model', 'res_id', 'create_uid', 'public', 'res_field'])
-        self.env.cr.execute('SELECT id, res_model, res_id, create_uid, public, res_field FROM ir_attachment WHERE id IN %s', [tuple(remaining.ids)])
-        for att_id, res_model, res_id, create_uid, public, res_field in self.env.cr.fetchall():
-            if public and operation == 'read':
+        prefetch_fields = ('res_model', 'res_id', 'create_uid', 'public', 'res_field')
+        for attachment in remaining.sudo().with_context(prefetch_fields=prefetch_fields):
```

Comment:
Doesn't exist prefetch_fields with list of fields. Only True or False.
Use fetch instead

Cache pollution? Maybe invalidate forbidden until we fix the cache pollution.